src/scripts/main.js

```diff
@@ -1 +1,89 @@
 'use strict';
+
+function createMessage(text, className = `success`) {
+  const div = document.createElement('div');
+
+  div.classList = className;
+  div.setAttribute('data-qa', 'notification');
+  div.textContent = text;
+  document.body.append(div);
+}
+
+function addClickHandlers(mouseClick) {
+  document.addEventListener('click', mouseClick);
+  document.addEventListener('contextmenu', mouseClick);
+}
+
+function removeClickHandlers(mouseClick) {
+  document.removeEventListener('click', mouseClick);
+  document.removeEventListener('contextmenu', mouseClick);
+}
+
+const firstPromise = new Promise((resolve, reject) => {
+  const leftClick = (e) => {
+    if (e.button === 0) {
+      resolve(`First promise was resolved`);
+      document.removeEventListener('click', leftClick);
+    }
+  };
+
+  document.addEventListener('click', leftClick);
+
+  setTimeout(() => {
+    reject(new Error(`First promise was rejected`));
+    document.removeEventListener('click', leftClick);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const mouseClick = (e) => {
+    if (e.button === 2) {
+      e.preventDefault();
+    }
+
+    if (e.button === 0 || e.button === 2) {
+      resolve(`Second promise was resolved`);
+    }
```

Comment:
The logic here resolves the second promise on either a left or right click. However, since the `removeClickHandlers` function is called immediately after, it might prevent the promise from resolving correctly if the event listener is removed before the promise is resolved. Consider restructuring the logic to ensure the promise resolves before removing the event listeners.